src/optimize/base_optimizer.js

```diff
@@ -101,6 +98,23 @@ export default class BaseOptimizer {
       });
     }
 
+    function maybeAddCacheLoader(uiBundles, cacheName, loaders) {
+      // only use cache-loader in dev mode
```

Comment:
In general I prefer comments that describe _why_ we're doing something, not _what_ is being done. E.g. it's obvious from the line below that we're only adding the cache loader in dev mode, but it's not clear why.

I suggest something like:

```js
/**
 * Adds a cache loader if we're running in dev mode. The reason we're not adding
 * the cache-loader when running in production mode is that it creates cache
 * files in optimize/.cache that are not necessary for distributable versions
 * of Kibana and just make compressing and extracting it more difficult.
 */
function maybeAddCacheLoader(uiBundles, cacheName, loaders) {
```

Now I don't have to know the details of the `cache-loader` to understand why it's excluded in production (in general you'd think caching is something we _want_ in production, so this helps clarify why it's _not_ something we want)

Reply:
Alright, that makes total sense. Thanks!